pkg/log/log.go

```diff
@@ -149,6 +158,82 @@ func (l *OsqueryLogAdapter) logInfoAboutUnrecognizedProcessLockingPidfile(p []by
 	level.Debug(l.logger).Log(append(processInfo, "msg", "detected non-osqueryd process using pidfile")...)
 }
 
+// runAndLogPs runs ps filtering on the given PID, and logs the output.
+func (l *OsqueryLogAdapter) runAndLogPs(pid int) {
+	if runtime.GOOS == "windows" {
+		return
+	}
+
+	pidStr := strconv.Itoa(pid)
+	cmd := exec.Command("ps", "-p", pidStr, "-o", "user,pid,ppid,pgid,stat,time,command")
```

Comment:
Do you think this makes more sense than `ps -eaf` or `ps aux`? I don't know how macOS handles the threads here.

Reply:
-e and -a seemed incompatible with filtering by PID with -p, but I might've missed something!

Re threads, `H/-L` don't seem to be doing it on macOS for me.